addOns/graphql/src/test/java/org/zaproxy/addon/graphql/VariantGraphQIUnitTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.graphql;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.testutils.TestUtils;
+
+public class VariantGraphQIUnitTest extends TestUtils {
+
+    @Test
+    void shouldNotExtractQueryIfHttpMessageIsNull() {
+        // given
+        VariantGraphQl vargraphqi = new VariantGraphQl();
+        // when
+        HttpMessage httpMsg = null;
+        // then
+        assertThrows(NullPointerException.class, () -> vargraphqi.setMessage(httpMsg));
+    }
+
+    @Test
+    void shouldNotExtractQueryForPostIfBodyIsEmptyAndNoContentTypeIsSet()
+            throws HttpMalformedHeaderException {
+        // given
+        VariantGraphQl vargraphqi = new VariantGraphQl();
+        // when
+        HttpRequestHeader httpReqHeader = new HttpRequestHeader();
+        httpReqHeader.setMessage("POST /abc/xyz HTTP/1.1");
+        HttpMessage httpMsg = new HttpMessage(httpReqHeader);
+        // then
+        assertThrows(NullPointerException.class, () -> vargraphqi.setMessage(httpMsg));
```

Comment:
Why is this case still leading to `NullPointerException`? It should assert that no parameters were extracted not the exception.

Reply:
The corrected condition doesn't throw a `NullPointerException` but the `else if` condition throws a `NullPointerException`.
While testing:
```
VariantGraphQlUnitTest > shouldNotExtractParametersForPostIfBodyIsEmptyAndNoContentTypeIsSet() FAILED
    java.lang.NullPointerException: Cannot invoke "String.contains(java.lang.CharSequence)" because "contentTypeHeader" is null
        at org.zaproxy.addon.graphql.VariantGraphQl.getQuery(VariantGraphQl.java:92)
        at org.zaproxy.addon.graphql.VariantGraphQl.setMessage(VariantGraphQl.java:46)
```

Code that is causing the Exception:
```
 else if (contentTypeHeader.contains("application/graphql")) {
                query = body;
            }
```
Shall I modify that condition too so that no 'NullPointerException` is thrown?

Reply:
Yes, no exception should be thrown. I think it could just return earlier (before checking the content-type) when POST and the body is empty.